modules/core/src/main/java/org/apache/ignite/internal/util/FastTimestamps.java

```diff
@@ -31,23 +35,15 @@ public class FastTimestamps {
     }
 
     private static void startUpdater() {
-        Thread updater = new Thread("FastTimestamps updater") {
-            /** {@inheritDoc} */
-            @Override
-            public void run() {
-                while (true) {
-                    coarseCurrentTimeMillis = System.currentTimeMillis();
-                    try {
-                        Thread.sleep(UPDATE_INTERVAL_MS);
-                    } catch (InterruptedException e) {
-                        break;
-                    }
-                }
-            }
-        };
+        ScheduledExecutorService scheduledExecutor = Executors.newSingleThreadScheduledExecutor(r -> {
+            Thread t = new Thread(r, "FastTimestamps updater");
```

Comment:
Can we use `IgniteThreadFactory`?

Reply:
Or maybe not. Node name is unknown anyway, and this thread belongs to JVM, not Ignite node

Reply:
you can create daemon threads with IgniteThreadFactory. Agree with the node name, but I actually think that we should make this an Ignite component: it will allow us to have the node name and to shutdown the executor gracefully. This is out of scope of this PR, though